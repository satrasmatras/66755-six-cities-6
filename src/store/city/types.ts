import City from "../../models/city";

export const CHANGE_CITY = `city/changeCity`;

export interface ChangeCityAction {
  type: typeof CHANGE_CITY,
  payload: City
}

export interface CityState {
  city: City
}

export type CityActionTypes = ChangeCityAction;
