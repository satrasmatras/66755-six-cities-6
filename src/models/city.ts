interface City {
  location: {
    latitude: number,
    longitude: number,
    zoom: number
  },
  name: string,
}

export default City;
