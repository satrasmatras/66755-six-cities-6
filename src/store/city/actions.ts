import {CHANGE_CITY, CityActionTypes} from "./types";
import City from "../../models/city";
import {Action, ActionCreator} from "redux";

export const changeCity: ActionCreator<Action> = (newCity: City): CityActionTypes => {
  return {
    type: CHANGE_CITY,
    payload: newCity
  };
};
