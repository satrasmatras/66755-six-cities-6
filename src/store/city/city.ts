import {CITIES} from "../../mocks/cities";
import {createAction, createReducer, createSlice} from "@reduxjs/toolkit";
import City from "../../models/city";

export interface CityState {
  city: City,
}

export const initialState: CityState = {
  city: CITIES[0],
};

export const CHANGE_CITY = `city/changeCity`;
export const changeCity = createAction<City>(CHANGE_CITY);

const cityReducer = createReducer(initialState, (builder) => {
  builder.addCase(changeCity, ((state, action) => {
    state.city = action.payload;
  }));
});

export default cityReducer;
