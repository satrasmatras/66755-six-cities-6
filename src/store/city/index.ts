import cityReducer from './slice';

export default cityReducer;
