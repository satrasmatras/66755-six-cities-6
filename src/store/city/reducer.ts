import {
  CHANGE_CITY,
  CityActionTypes,
  CityState
} from "./types";
import {CITIES} from "../../mocks/cities";

const initialState: CityState = {
  city: CITIES[0],
};

export const cityReducer = (state = initialState, action: CityActionTypes): CityState => {
  switch (action.type) {
    case CHANGE_CITY:
      return {
        ...state,
        city: action.payload
      };
    default:
      return state;
  }
};
