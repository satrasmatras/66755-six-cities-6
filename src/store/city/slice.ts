import {CITIES} from "../../mocks/cities";
import {createSlice} from "@reduxjs/toolkit";
import City from "../../models/city";

export interface CityState {
  city: City,
}

const initialState: CityState = {
  city: CITIES[0],
};

const citySlice = createSlice({
  name: `city`,
  initialState: initialState,
  reducers: {
    changeCity: (state, action) => {
      state.city = action.payload;
    }
  }
});

export const {
  changeCity,
} = citySlice.actions;

export default citySlice.reducer;
