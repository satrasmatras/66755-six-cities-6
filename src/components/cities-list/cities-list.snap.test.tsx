import React from "react";
import {render} from "@testing-library/react";
import {Router} from "react-router-dom";
import browserHistory from "../../browser-history";
import CitiesList from "./cities-list";
import {Provider} from "react-redux";
import {initialState} from "../../store/city/city";
import configureStore from "redux-mock-store";

const store = configureStore(undefined)({city: initialState});

it(`Should cities list render correctly`, () => {
  const {container} = render(
      <Provider store={store}>
        <Router history={browserHistory}>
          <CitiesList />
        </Router>
      </Provider>
  );
  expect(container).toMatchSnapshot();
});


