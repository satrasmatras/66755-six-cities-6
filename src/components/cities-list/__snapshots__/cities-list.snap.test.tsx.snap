// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should cities list render correctly 1`] = `
<div>
  <ul
    class="locations__list tabs__list"
  >
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item tabs__item--active"
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item "
        href="#"
      >
        <span>
          Cologne
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item "
        href="#"
      >
        <span>
          Brussels
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item "
        href="#"
      >
        <span>
          Amsterdam
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item "
        href="#"
      >
        <span>
          Hamburg
        </span>
      </a>
    </li>
    <li
      class="locations__item"
    >
      <a
        class="locations__item-link tabs__item "
        href="#"
      >
        <span>
          Dusseldorf
        </span>
      </a>
    </li>
  </ul>
</div>
`;
